planner/core/explain.go

```diff
@@ -696,8 +700,44 @@ func (p *TiKVSingleGather) ExplainInfo() string {
 
 // ExplainInfo implements Plan interface.
 func (p *PhysicalMemTable) ExplainInfo() string {
-	if !metricschema.IsMetricTable(p.Table.Name.L) {
+	return genMetricTableExplainInfo(p.ctx, p.DBName, p.Table, p.Extractor)
+}
+
+func genMetricTableExplainInfo(ctx sessionctx.Context, dbName model.CIStr, tblInfo *model.TableInfo, extractor MemTablePredicateExtractor) string {
+	if dbName.L != util.MetricSchemaName.L || !metricschema.IsMetricTable(tblInfo.Name.L) {
+		return ""
+	}
```

Comment:
I think this condition can be eliminated because of all metrics table contains the `MetricTableExtractor`.

Reply:
great.